shell/platform/darwin/ios/framework/Source/FlutterKeyboardManager.mm

```diff
@@ -0,0 +1,117 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterKeyboardManager.h"
+#include "flutter/fml/memory/weak_ptr.h"
+
+@interface FlutterKeyboardManager ()
+
+/**
+ * The owner set by initWithOwner.
+ */
+@property(nonatomic) fml::WeakPtr<UIResponder> owner;
+
+/**
+ * The primary responders added by addPrimaryResponder.
+ */
+@property(nonatomic) NSMutableArray<id<FlutterKeyPrimaryResponder>>* primaryResponders;
+
+/**
+ * The secondary responders added by addSecondaryResponder.
+ */
+@property(nonatomic) NSMutableArray<id<FlutterKeySecondaryResponder>>* secondaryResponders;
+
+- (void)dispatchToSecondaryResponders:(nonnull UIPress*)press
+                            withEvent:(nonnull UIPressesEvent*)event
+                             complete:(nonnull KeyEventCompleteCallback)callback
+    API_AVAILABLE(ios(13.4));
+
+@end
+
+@implementation FlutterKeyboardManager
+
+- (nonnull instancetype)initWithOwner:(fml::WeakPtr<UIResponder>)weakOwner {
+  self = [super init];
+  if (self != nil) {
+    _owner = weakOwner;
+    _primaryResponders = [[NSMutableArray alloc] init];
+    _secondaryResponders = [[NSMutableArray alloc] init];
+  }
+  return self;
+}
+
+- (void)addPrimaryResponder:(nonnull id<FlutterKeyPrimaryResponder>)responder {
+  [_primaryResponders addObject:responder];
+}
+
+- (void)addSecondaryResponder:(nonnull id<FlutterKeySecondaryResponder>)responder {
+  [_secondaryResponders addObject:responder];
+}
+
+- (void)handlePresses:(nonnull NSSet<UIPress*>*)presses
+            withEvent:(nonnull UIPressesEvent*)event
+             complete:(nonnull KeyEventCompleteCallback)completeCallback API_AVAILABLE(ios(13.4)) {
+  if (@available(iOS 13.4, *)) {
+    // no-op
+  } else {
+    return;
+  }
+
+  // Having no primary responders requires extra logic, but Flutter hard-codes
+  // all primary responders, so this is a situation that Flutter will never
+  // encounter.
+  NSAssert([_primaryResponders count] >= 0, @"At least one primary responder must be added.");
+
+  __unsafe_unretained __typeof__(self) weakSelf = self;
```

Comment:
Is this `__unsafe_unretained` due to the reason you mentioned about `__block` being unsupported?

Reply:
No, it's because __weak is not supported. Actually, I thought I got rid of most of these and replaced them with fml::WeakPtr.  I'll see if I can do that here.